core/src/test/java/org/infinispan/notifications/cachelistener/CacheNotifierImplWithConverterTest.java

```diff
@@ -0,0 +1,322 @@
+import org.infinispan.compat.TypeConverter;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.container.InternalEntryFactory;
+import org.infinispan.container.entries.ImmortalCacheEntry;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.infinispan.context.InvocationContext;
+import org.infinispan.context.impl.NonTxInvocationContext;
+import org.infinispan.distribution.DistributionManager;
+import org.infinispan.filter.NamedFactory;
+import org.infinispan.interceptors.impl.WrappedByteArrayConverter;
+import org.infinispan.interceptors.locking.ClusteringDependentLogic;
+import org.infinispan.lifecycle.ComponentStatus;
+import org.infinispan.metadata.Metadata;
+import org.infinispan.notifications.cachelistener.annotation.CacheEntryCreated;
+import org.infinispan.notifications.cachelistener.annotation.CacheEntryExpired;
+import org.infinispan.notifications.cachelistener.annotation.CacheEntryModified;
+import org.infinispan.notifications.cachelistener.annotation.CacheEntryRemoved;
+import org.infinispan.notifications.cachelistener.cluster.ClusterEventManager;
+import org.infinispan.notifications.cachelistener.event.CacheEntriesEvictedEvent;
+import org.infinispan.notifications.cachelistener.event.CacheEntryActivatedEvent;
+import org.infinispan.notifications.cachelistener.event.CacheEntryCreatedEvent;
+import org.infinispan.notifications.cachelistener.event.CacheEntryEvent;
+import org.infinispan.notifications.cachelistener.event.CacheEntryExpiredEvent;
+import org.infinispan.notifications.cachelistener.event.CacheEntryInvalidatedEvent;
+import org.infinispan.notifications.cachelistener.event.CacheEntryLoadedEvent;
+import org.infinispan.notifications.cachelistener.event.CacheEntryModifiedEvent;
+import org.infinispan.notifications.cachelistener.event.CacheEntryPassivatedEvent;
+import org.infinispan.notifications.cachelistener.event.CacheEntryRemovedEvent;
+import org.infinispan.notifications.cachelistener.event.CacheEntryVisitedEvent;
+import org.infinispan.notifications.cachelistener.event.Event;
+import org.infinispan.notifications.cachelistener.event.TransactionCompletedEvent;
+import org.infinispan.notifications.cachelistener.event.TransactionRegisteredEvent;
+import org.infinispan.notifications.cachelistener.filter.CacheEventConverter;
+import org.infinispan.notifications.cachelistener.filter.CacheEventConverterFactory;
+import org.infinispan.notifications.cachelistener.filter.EventType;
+import org.infinispan.test.AbstractInfinispanTest;
+import org.infinispan.test.fwk.TestInternalCacheEntryFactory;
+import org.infinispan.transaction.xa.GlobalTransaction;
+import org.mockito.Mockito;
+import org.mockito.stubbing.Answer;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+
+@Test(groups = "unit", testName = "notifications.cachelistener.CacheNotifierImplWithConverterTest")
+public class CacheNotifierImplWithConverterTest extends AbstractInfinispanTest {
```

Comment:
I think this would be a lot cleaner if instead we extend `CacheNotifierImplTest` and just passed an argument or callback to guarantee equality on the parts that were different. Otherwise we have a ton of duplicated code.